### DISCUSSION TITLE
v0.6.0

### DISCUSSION THREAD

**Discussion by sonichi** [microsoft/FLAML#175] [Show and tell]
In this release, we added support for time series forecasting task and NLP model fine tuning. Also, we have made a large number of feature & performance improvements.

- data split by 'time' for time-ordered data, and by 'group' for grouped data.
- support parallel trials and random search in `AutoML.fit()` API.
- support warm-start in `AutoML.fit()` by using previously found start points.
- support constraints on training/prediction time per model.
- new optimization metric: ROC_AUC for multi-class classification, MAPE for time series forecasting. 
- utility functions for getting normalized confusion matrices and multi-class ROC or precision-recall curves.
- automatically retrain models after search by default; options to disable retraining or enforce time limit.
- CFO supports hierarchical search space and uses points_to_evaluate more effectively.
- variation of CFO optimized for unordered categorical hps.
- BlendSearch improved for better performance in parallel setting.
- memory overhead optimization.
- search space improvements for random forest and lightgbm.
- make stacking ensemble work for categorical features.
- python 3.9 support.
- experimental support for automated fine-tuning of transformer models from huggingface.
- experimental support for time series forecasting.
- warnings to suggest increasing time budget, and warning to inform users there is no performance improvement for a long time.

Minor updates
- make log file name optional.
- notebook for time series forecasting.
- notebook for using AutoML in sklearn pipeline.
- bug fix when training_function returns a value.
- support fixed random seeds to improve reproducibility.
- code coverage improvement.
- exclusive upper bounds for hyperparameter type randint and lograndint.
- experimental features in BlendSearch.
- documentation improvement.
- bug fixes for multiple logged metrics in cv.
- adjust epsilon when time per trial is very fast.

Contributors
- @sonichi 
- @qingyun-wu 
- @int-chaos 
- @liususan091219 
- @Yard1 
- @bnriiitb
- @su2umaru 
- @eduardobull
- @sek788432
- @ekzhu 
- @anshumandutt 
- @yue-msr 
- @sadtaf 
- @fzanartu 
- @dsbyprateekg
- @hanhanwu 
- @PardeepRassani 
- @gianpDomiziani 
- @stepthom
- @anhnht3
- @zzheng93
- @flippercy
- @luizhemelo
- @nabalamu
- @lostmygithubaccount 
- @suryajayaraman

<hr /><em>This discussion was created from the release <a href='https://github.com/microsoft/FLAML/releases/tag/v0.6.0'>v0.6.0</a>.</em>
